system/jlib/jtrace.hpp

```diff
@@ -15,11 +15,47 @@
     limitations under the License.
 ############################################################################## */
 
-
-
 #ifndef JTRACE_HPP
 #define JTRACE_HPP
 
+class jlib_decl LogTrace
+{
+private:
+    StringAttr   globalId;
+    StringAttr   callerId;
+    StringAttr   localId;
+
+    StringAttr   globalIdHTTPHeaderName = "HPCC-Global-Id";
+    StringAttr   callerIdHTTPHeaderName = "HPCC-Caller-Id";
+
+    const char* assignLocalId();
+
+public:
+
+    LogTrace() {};
+    LogTrace(const char * globalId);
+
+    const char* queryGlobalId() const;
+    const char* queryCallerId() const;
+    const char* queryLocalId() const;
+    const char* queryGlobalIdHTTPHeaderName() const { return globalIdHTTPHeaderName.get(); }
+    const char* queryCallerIdHTTPHeaderName() const { return callerIdHTTPHeaderName.get(); }
+
+    void setHttpIdHeaderNames(const char *global, const char *caller)
+    {
+        setGlobalIdHTTPHeaderName(global);
+        setCallerIdHTTPHeaderName(caller);
+    }
+
+    void setGlobalIdHTTPHeaderName(const char* headerName) { if (!isEmptyString(headerName)) globalIdHTTPHeaderName.set(headerName);};
```

Comment:
Should be possible to make this private (and delete it if it is only called from setHttpIdHeaderNames)